### PR TITLE
Use named constants for special offsets

### PR DESCRIPTION
**Pull request by jnjackins**
Note: This PR breaks backward compatibility.

This change replaces hardcoded -1 and -2 offsets with named constants, and fixes them to be consistent with other libraries / kafka docs.

For #120.

### CODE REVIEW

File: reader.go [hunk 1]
```diff
@@ -17,8 +17,8 @@ import (
 )
 
 const (
```
Comment (jnjackins) [inline reader.go]: I reversed these. Sarama and kafkacat both treat them this way (last == -1 and first == -2). Neither seem to be mentioned in the Kafka protocol, so I'm unclear whether this is a library convention or a fundamental Kafka thing.

**Comment by jnjackins** [comment]
More data: the kafka python library agrees with sarama:
```
class OffsetResetStrategy(object):
    LATEST = -1
    EARLIEST = -2
    NONE = 0
```

File: reader.go (hunk 1 above)
Comment (abraithwaite) [reply reader.go]: FWIW, I prefer 'Newest' and 'Oldest', but being in line with other libraries is probably the most important.

**Comment by achille-roussel** [comment]
@jnjackins would you have time to fix the tests? Or maybe someone else can take a crack at it?

**Comment by jnjackins** [comment]
I'll look at it again today and let you know if I need help, how does that sound. :)

**Pull request merged**